src/components/DashboardEditor/DashboardEditor.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { settings } from '../../constants/Settings';
+import { CARD_TYPES } from '../../constants/LayoutConstants';
+import { DashboardGrid, CardEditor } from '../../index';
+
+import DashboardEditorHeader from './DashboardEditorHeader/DashboardEditorHeader';
+import { getDefaultCard, getCardPreview } from './editorUtils';
+
+const { iotPrefix } = settings;
+
+export const defaultI18N = {
+  headerImportButton: 'Import',
+  headerExportButton: 'Export',
+  headerCancelButton: 'Cancel',
+  headerSubmitButton: 'Save and close',
+  galleryHeader: 'Gallery',
+  openGalleryButton: 'Open gallery',
+  closeGalleryButton: 'Back',
+  openJSONButton: 'Open JSON editor',
+};
+
+const defaultProps = {
+  initialValue: {
+    cards: [],
+    layouts: {},
+  },
+  supportedCardTypes: [CARD_TYPES.BAR, CARD_TYPES.TIMESERIES, CARD_TYPES.VALUE, CARD_TYPES.TABLE],
+  renderHeader: null,
+  renderCardPreview: () => null,
+  headerBreadcrumbs: null,
+  onImport: null,
+  onExport: null,
+  i18n: defaultI18N,
+};
+
+const propTypes = {
+  /** Dashboard title */
+  title: PropTypes.string.isRequired,
+  /** initial dashboard data to edit */
+  initialValue: PropTypes.shape({
+    cards: PropTypes.array,
+    layouts: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+  }),
+  /** supported card types */
+  supportedCardTypes: PropTypes.arrayOf(PropTypes.string),
+  /** if provided, renders header content above preview */
+  renderHeader: PropTypes.func,
+  /** if provided, is used to render cards in dashboard */
+  renderCardPreview: PropTypes.func,
+  /** if provided, renders array elements inside of BreadcrumbItem in header */
+  headerBreadcrumbs: PropTypes.arrayOf(PropTypes.element),
+  /** if provided, renders import button linked to this callback */
+  onImport: PropTypes.func,
```

Comment:
For each callback, could you add what will be given back to the callback as arguments?

Reply:
fixed